s2/bq/doc/REFERENCE.md

```diff
@@ -0,0 +1,162 @@
+## Reference
+
+### S2
+
+#### VERSION
+
+{{% bannerNote type="code" %}}
+s2.VERSION()
+{{%/ bannerNote %}}
+
+Returns the current version of the s2 library.
+
+#### GEOJSONBOUNDARY_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMKEY(key STRING) 
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners of a quadkey.
+
+* `key`: `STRING` quadkey to extract the corners from.
+
+#### GEOJSONBOUNDARY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.GEOJSONBOUNDARY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Returns a GeoJSON containing the boundary corners given a longitud, latitud and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### ID_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Convert from hilbert quadtree id to s2 cell id.
+
+* `key`: `STRING` quadkey to be converted.
+
+#### ID_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.ID_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the s2 cell id for a given longitud, latitud and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### KEY_FROMID
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Convert from s2 cell id to hilbert quadtree id.
+
+* `id`: `INT64` s2 cell id to be converted.
+
+#### KEY_FROMLONGLAT
+
+{{% bannerNote type="code" %}}
+s2.KEY_FROMLONGLAT(longitude FLOAT64, latitude FLOAT64, level NUMERIC)
+{{%/ bannerNote %}}
+
+Generate the quadkey for a given longitud, latitud and zoom level.
+
+* `longitude`: `FLOAT64` horizontal coordinate on the map.
+* `latitude`: `FLOAT64` vertical coordinate on the map.
+* `level`: `NUMERIC` Level of detail or zoom.
+
+#### LONG_FROMID
+
+{{% bannerNote type="code" %}}
+s2.LONG_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Extract the longitude component from a give s2 cell id.
+
+* `id`: `INT64` s2 cell id.
+
+#### LONG_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.LONG_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Extract the longitude component from a quadkey.
+
+* `key`: `STRING` quadkey.
+
+#### LAT_FROMID
+
+{{% bannerNote type="code" %}}
+s2.LAT_FROMID(id INT64)
+{{%/ bannerNote %}}
+
+Extract the latitude component from a give s2 cell id.
+
+* `id`: `INT64` s2 cell id.
+
+#### LAT_FROMKEY
+
+{{% bannerNote type="code" %}}
+s2.LAT_FROMKEY(key STRING)
+{{%/ bannerNote %}}
+
+Extract the latitude component from a quadkey.
+
+* `key`: `STRING` quadkey.
+
+#### LONGLAT_ASS2
```

Comment:
Yes, I'm not super fan either of the double S but where using the same notation in all the other modules LONGLAT_ASH3, LONGLAT_ASQUADINT etc..
We rarely use more than one underscore in the names as well, mainly when it's a ST_XXXX_POINT/BOUNDARY.

What about, S2_FROMLONGLAT?

Reply:
Something like `S2_FROMLONGLAT` SGTM. You might consider using `ID_FROMLONGLAT` or `LONGLAT_ASID` since in practice these calls will be prefixed with `<package>.s2.` (right?).

Reply:
Yep, naming is troublesome here. We wanted to be consistent with BigQuery names so the stardard would be `LONGLAT_ASS2` and `ST_ASS2` which isn't great. We can absolutely change it (and probably should) but we should be consistent and also follow whatever standard we define here for the other modules too